pandas/core/missing.py

```diff
@@ -620,144 +618,65 @@ def interpolate_2d(
     ----------
     values: array-like
         Input array.
-    method: str, default "pad"
-        Interpolation method. Could be "bfill" or "pad"
+    method: {"pad", "backfill", "ffill", "bfill"}, default "pad"
+        Interpolation method.
     axis: 0 or 1
         Interpolation axis
     limit: int, optional
         Index limit on interpolation.
     limit_area: str, optional
-        Limit area for interpolation. Can be "inside" or "outside"
+        Limit area for interpolation. Can be "inside" or "outside".
+    mask: numpy array, optional
+        boolean array of values to be interpolated.
 
     Returns
     -------
-    values: array-like
-        Interpolated array.
+    tuple of values: array-like, mask: array or None
+        Interpolated array and updated mask.
     """
     if limit_area is not None:
-        return np.apply_along_axis(
-            partial(
-                _interpolate_with_limit_area,
-                method=method,
-                limit=limit,
-                limit_area=limit_area,
+        return (
+            np.apply_along_axis(
+                partial(
+                    _interpolate_with_limit_area,
+                    method=method,
+                    limit=limit,
+                    limit_area=limit_area,
+                ),
+                axis,
+                values,
             ),
-            axis,
-            values,
+            mask,
         )
 
-    orig_values = values
-
-    transf = (lambda x: x) if axis == 0 else (lambda x: x.T)
+    if not np.all(values.shape):
```

Comment:
`not values.size`

Reply:
that was already in the code (in _pad_2d and _backfill_2d). do you want it changed?

Reply:
not a big deal